FuelSDK-Test/FuelSDKConfigurationSectionTest.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            Assert.That(() => GetCustomConfigurationSectionFromConfigFile(missingRequiredClientIdConfigFileName), Throws.TypeOf<ConfigurationErrorsException>());
+        }
+
+        [Test()]
+        public void MissingRequiredClientSecretPropertyFromConfigSection()
+        {
+            Assert.That(() => GetCustomConfigurationSectionFromConfigFile(missingRequiredClientSecretConfigFileName), Throws.TypeOf<ConfigurationErrorsException>());
+        }
+
+        [Test()]
+        public void MissingSoapEndPointPropertyFromConfigSection()
+        {
+            FuelSDKConfigurationSection section = GetCustomConfigurationSectionFromConfigFile(requiredPropertiesOnlyConfigFileName);
+            var attribute = section.GetType().GetProperty("SoapEndPoint").GetCustomAttributes(typeof(ConfigurationPropertyAttribute), false).Single() as ConfigurationPropertyAttribute;
+            Assert.AreEqual(section.SoapEndPoint, attribute.DefaultValue);
+        }
+
+        [Test()]
+        public void MissingAuthEndPointPropertyFromConfigSection()
+        {
+            FuelSDKConfigurationSection section = GetCustomConfigurationSectionFromConfigFile(requiredPropertiesOnlyConfigFileName);
+            var attribute = section.GetType().GetProperty("AuthenticationEndPoint").GetCustomAttributes(typeof(ConfigurationPropertyAttribute), false).Single() as ConfigurationPropertyAttribute;
+            Assert.AreEqual(section.AuthenticationEndPoint, attribute.DefaultValue);
+        }
+
+        [Test()]
+        public void MissingRestEndPointPropertyFromConfigSection()
+        {
+            FuelSDKConfigurationSection section = GetCustomConfigurationSectionFromConfigFile(requiredPropertiesOnlyConfigFileName);
+            var attribute = section.GetType().GetProperty("RestEndPoint").GetCustomAttributes(typeof(ConfigurationPropertyAttribute), false).Single() as ConfigurationPropertyAttribute;
+            Assert.AreEqual(section.RestEndPoint, attribute.DefaultValue);
+        }
+
+        [Test()]
+        public void AllPropertiesSetInConfigSection()
+        {
+            FuelSDKConfigurationSection section = GetCustomConfigurationSectionFromConfigFile(allPropertiesSetConfigFileName);
+            Assert.AreEqual(section.AppSignature, "none");
+            Assert.AreEqual(section.ClientId, "abc");
+            Assert.AreEqual(section.ClientSecret, "cde");
+            Assert.AreEqual(section.SoapEndPoint, "https://soapendpoint.com");
+            Assert.AreEqual(section.AuthenticationEndPoint, "https://authendpoint.com");
+            Assert.AreEqual(section.RestEndPoint, "https://restendpoint.com");
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (File.Exists(configFilePath))
+            {
+                File.Delete(configFilePath);
+            }
+        }
+
+        private void CreateConfigFileFromResource(string resourceFileName)
```

Comment:
is there a reason to read and write the config file again? I guess my question is "We have the config files stored under the configFiles folder in the repo. why not directly use them instead of reading it and writing it again?

Please correct me if my understanding is wrong.

Reply:
I added this as a mean to test the behavior of the custom config section retrieval for different variations of the config file, since the ETClient default constructor takes the config values from the App.config file.

Reply:
@sfdrogojan lets talk about this during tomorrow's sync up meeting